core/utils/compare.go

```diff
@@ -0,0 +1,9 @@
+package utils
+
+func IsZero[C comparable](val C) bool {
+	var zero C
+	return zero == val
+}
+func Equal[C comparable](val, other C) bool {
```

Comment:
Just curious.
If this function requires comparable types, and we have 2 comparable types, why can't we just directly compare them via ==?
Why need this function?